src/utils/props.js

```diff
@@ -91,10 +91,17 @@ export function applyDefaultProps(instance, oldProps, newProps) {
 
   // update event handlers
   if (!newProps.ignoreEvents) {
-    eventHandlers.forEach(function(evt) {
-      isFunction(oldProps[evt], instance.removeListener) && instance.removeListener(evt, oldProps[evt])
-      isFunction(newProps[evt], instance.on) && instance.on(evt, newProps[evt])
-    })
+    for (let i = 0; i < eventHandlers.length; i++) {
+      const evt = eventHandlers[i]
+      if (oldProps[evt] !== newProps[evt]) {
+        if (typeof oldProps[evt] === 'function' && typeof instance.removeListener === 'function') {
```

Comment:
Do this check once outside of the loop and cache the result: `typeof instance.removeListener === 'function'`
```
let hasRemoveListener = typeof instance.removeListener === 'function'
for (let i = 0; i < eventHandlers.length; i++) {
   ...
   if (typeof oldProps[evt] === 'function' && hasRemoveListener) { ...}
```

Reply:
Yeah I like the idea but it there anytime when you don't get `on` and `removeListener`? Can the `instance` sometimes don't extend EventEmitter?

Reply:
I also would expect this. And I think it would be save, because there is an explicit check that the instance is of type `PIXI.DisplayObject` which extends `EventEmitter`: https://github.com/pixijs/pixi.js/blob/5ceb18ad00289521023f81e6cede9b6cc1c7d377/packages/display/src/DisplayObject.js